src/function/open-referral-function/src/FamilyHubs.OpenReferral.Function/ClientServices/HsdaApiService.cs

```diff
@@ -49,7 +49,17 @@ public class HsdaApiService(ILogger<HsdaApiService> logger, HttpClient httpClien
                 continue;
             }
 
-            servicesById.Add(new ServiceJson (Id: serviceId, Json: jsonResponse!));
+            try
+            {
+                Service? service = JsonSerializer.Deserialize<Service>(jsonResponse!);
+                servicesById.Add(service ?? throw new InvalidOperationException());
```

Comment:
Do we need to throw an exception here if we cannot deserialize it? I see that it will get caught by the catch and logged but using exceptions for process flow can introduce performance issues, especially if this happens a lot and it is also harder to read.

Might be better to null check the service variable and log a specific error that the deserialization returned null

Reply:
Good point, implemented in: [e886d7d](https://github.com/DFE-Digital/fh-services/pull/92/commits/e886d7d9e4fb21d70148b3e7e68c2afe9d87240e)